### ISSUE TITLE
PyRight doesn't see arguments of constructor for class inherited from pandas.DataFrame

### ISSUE THREAD

**Issue by upginius** [microsoft/pyright#3435]
**Describe the bug**
PyRight doesn't see arguments of constructor for a class that is inherited from pandas.DataFrame.

**To Reproduce**
```python
class MyDataFrame(pd.DataFrame):
    def __init__(self, extra_attr: str):
        super().__init__()
        self._extra_attr = extra_attr


test_df = MyDataFrame(extra_attr="test")
```
Got problem: `No parameter named "extra_attr"`

**Expected behavior**
No problems reported.

**VS Code extension or command-line**
Pylance v2022.4.3

**Comment by erictraut** [comment]
I'm not able to repro the problem in the sample above, and I don't have any theories about why you would see such an error in this case.

Can you verify that when you copy and paste the code above into a new source file that you still see this error?

**Comment by upginius** [comment]
Yes, it is reproduced in a new file under Python 3.9.12 and Pandas 1.4.2.

**Comment by JelleZijlstra** [comment]
I see this too in an env with latest pandas installed. The issue is that DataFrame has a conflicting `__new__` method. Pyright warns about this:

```
Mismatch between signature of __new__ and __init__ in class "MyDataFrame"
  Signature of __init__ is "(extra_attr: str)"
  Signature of __new__ is "(data: _ListLike | DataFrame | Dict[Any, Any] | None = ..., index: _ListLike | None = ..., columns: _ListLike | None = ..., dtype: ... = ..., copy: _bool = ...)"
```

**Comment by erictraut** [comment]
I'm still not able to repro this. I've installed pandas 1.4.2 in Python 3.9.12. Do you have any stubs installed for Pandas?

@JelleZijlstra, based on your analysis, this sounds like a bug in the stubs and/or the library? If the `__new__` method doesn't have a parameter named `extra_attr` and doesn't accept a `**kwargs`, then pyright is correct in flagging it as an error if a call to the constructor includes a named argument with the name `extra_attr`.

**Comment by JelleZijlstra** [comment]
I think it's the pandas stubs shipped with pylance. In my installed copy I see

```
class DataFrame(NDFrame, OpsMixin):
    _ListLike = Union[
        np.ndarray,
        List[Dtype],
        Dict[_str, _np.ndarray],
        Sequence,
        Index,
        Series,
    ]
    def __new__(
        cls,
        data: Optional[Union[_ListLike, DataFrame, Dict[Any, Any]]] = ...,
        index: Optional[Union[Index, _ListLike]] = ...,
        columns: Optional[_ListLike] = ...,
        dtype=...,
        copy: _bool = ...,
    ) -> DataFrame: ...
```

But in fact DataFrame doesn't override `__new__` at runtime, so the stubs are wrong.
